centroid_decompostition.cpp

```diff
@@ -0,0 +1,70 @@
+#include <iostream>
+#include <list>
+
+using namespace std;
+
+struct Graph 
+{
+	int V;
+	list<int> *adj;
+	int *size;
+	bool *deleted;
+
+	Graph (int num) : V(num) {
+		adj = new list<int>[V];
+		size = new int[V];
+		deleted = new bool[V];	
+	}
+
+	void addEdge (int x, int y);
+
+	void calcSizes (int cur, int par);
+
+	int findCentroid (int cur, int par, int vertices);
+
+	void decomposeTree (int cur, int total);
```

Comment:
This class needs  a destructor if it will use dynamic arrays so that no memory is leaked. A better option is std::vector.